snappy.cc

```diff
@@ -1066,13 +1117,18 @@ void MemCopy64(char* dst, const void* src, size_t size) {
     data = _mm256_lddqu_si256(static_cast<const __m256i *>(src) + 1);
     _mm256_storeu_si256(reinterpret_cast<__m256i *>(dst) + 1, data);
   }
+#elif defined(__aarch64__)
+  // Emperically it is faster to just copy all 64 rather than branching.
+  (void)kShortMemCopy;
+  (void)size;
+  FixedSizeMemMove<64>(dst, src);
 #else
-  std::memmove(dst, src, kShortMemCopy);
+  FixedSizeMemMove<kShortMemCopy>(dst, src);
   // Profiling shows that nearly all copies are short.
   if (SNAPPY_PREDICT_FALSE(size > kShortMemCopy)) {
-    std::memmove(dst + kShortMemCopy,
-                 static_cast<const uint8_t*>(src) + kShortMemCopy,
-                 64 - kShortMemCopy);
+    FixedSizeMemMove<kShortMemCopy>(
```

Comment:
[Q] We've already copied `kShortMemCopy` bytes on line 1126. This code used to copy the remaining  bytes if `size > kShortMemCopy` by copying `64 - kShortMemCopy` bytes.  The new code, however, always copies `kShortMemCopy`, assuming `2 * kShortMemCopy == 64`, right? If so, do we need to have a static assertion verifying this?

Reply:
It is defined as 32 on line 1100, 30 lines up. Adding a static assert that 32*2  == 64 seems silly. Also the original AVX impl was relying on this as well.